### PR TITLE
Published rate order

### PR DESCRIPTION
**Pull request by szerintedmi**
- Feature for orders which will match on price published by Augmint's Rates contract.  
  Order placed with 0 price will mark this type of order.
- Change match price logic: price is maker price (price of order placed earlier)

### CODE REVIEW

File: rinkeby_migrations/15_set_new_MonetarySupervisor_live.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+const FeeAccount = artifacts.require("./FeeAccount");
+
+module.exports = function(deployer, network) {
+    deployer.then(async () => {
+        // NB: web3.version.network throws error with truffle migrate --dry-run...
+        if (network === "rinkeby" || network === "rinkeby-fork") {
+            // Truffle artifacts are in unknown state when truffle migrate starts from this step on rinkeby.
+            // Therefore we can't use addresses from there.
+            // But this script will be run only ONCE on rinkeby so it's fine to hardcode addresses
+            const ratesAddress = "0xcA8100FCcb479516A5b30f8Bc5dAeA09Fb7a7473";
```
Comment (phraktle) [inline rinkeby_migrations/15_set_new_MonetarySupervisor_live.js]: Could you get these addresses from an Abiniser environment json?

File: contracts/Exchange.sol [hunk 2]
```diff
@@ -155,10 +169,15 @@ contract Exchange {
...
+        require(buy.price >= sell.price || buy.price == 0, "buy price must be >= sell price or sell or buy price == 0");
+
+        uint publishedRate;
+        (publishedRate, ) = rates.rates(augmintToken.peggedSymbol());
+        uint buyPrice = buy.price > 0 ? buy.price : publishedRate;
+        uint sellPrice = sell.price > 0 ? sell.price : publishedRate;
 
-        // meet in the middle
-        uint price = uint(buy.price).add(sell.price).div(2);
+        // pick maker's price (whoever placed order sooner considered as maker)
```
Comment (phraktle) [inline contracts/Exchange.sol]: What prompted this change? (One could argue for either approach)

Comment (szerintedmi) [reply contracts/Exchange.sol]: We discussed this a while back around [this trello card](https://trello.com/c/i2g31QCE/131-exchange-match-to-use-price-of-the-earlier-order-on-match), maybe on Discord. @krosza 's view was to go with this as this the standard. I did some research too and indeed seems to be de-facto that's the standard.

File: rinkeby_migrations/15_set_new_MonetarySupervisor_live.js (hunk 1 above)
Comment (szerintedmi) [reply rinkeby_migrations/15_set_new_MonetarySupervisor_live.js]: We could get the latest deployed address for a given abi. I would feel the script a bit more flaky and difficult to check before running compared to hardcoding address where we can check the addresses for a given deployment's on the under-the-hood page.

**Pull request merged**